### PR TITLE
fix: correctly print double escaped backslashes with no raw value

### PR DESCRIPTION
**Pull request by paoloricciuti**
Closes https://github.com/sveltejs/svelte/issues/15476

A bit of an edge case but a valid usecase nontheless.

### CODE REVIEW

File: test/quotes.test.js [hunk 1]
```diff
@@ -86,6 +86,42 @@ test('does not escape already-escaped single quotes', () => {
 	expect(code).toMatchInlineSnapshot(`"const str = 'a\\'b';"`);
 });
 
+test('correctly handle double escaped backslashes when theres no raw', () => {
+	// doing this since if there's a raw value it will prefer that
```
Comment (manuel3108) [inline test/quotes.test.js:90]: Maybe I'm misunderstanding your comment, but that's exactly why we have the clean function, that you are already calling. Haven't tested it, but I think you should be fine using `load()`

Comment (paoloricciuti) [reply test/quotes.test.js:90]: Mmm the test was passing even before the changes when I called load...will check tho

Comment (manuel3108) [reply test/quotes.test.js:90]: Interesting, you are right. But I don't know why :thinking: ....

I would have expected the following code to produce exactly the same AST as your manual way.
```js
const ast2 = load("var text = $.text('\\\\');");
```

But it's not. That's why you were unable to reproduce the issue.

Looks like acorn is doing some preprocessing while converting to the AST so that for whatever reason calling `load()` brings `\\` as the value for the literal instead of `\\\\`.

So changing the input to ...
```js
const ast = load("var text = $.text('\\\\\\\\');");
```

... produces the error you were trying to reproduce. And your fix still fixes the issue :laughing: 

That being said, I don't know what's less ugly to write in the test, I will leave that up to you

Comment (paoloricciuti) [reply test/quotes.test.js]: Yup just explored the same and updated the comment...the value generated by acorn is just `\\` while `raw` contains `\\\\`...I've even thought if we should match this in the svelte parse but tbf it feels more weird (maybe even buggish) what `acorn` is doing

**Comment by dummdidumm** [comment]
Investigated a similar bug and concluded the whole "we should not escape at this point" logic is flawed, we should just remove it - opened #58 as an alternative
